pkg/util/util.go

```diff
@@ -968,20 +982,20 @@ func Unzip(src, dest, pathToUnzip string) ([]string, error) {
 	return filenames, nil
 }
 
-// DownloadFile uses the url to download the file to the filepath
-func DownloadFile(url string, filepath string) error {
-	// Create the file
-	out, err := os.Create(filepath)
+// DownloadFile downloads the file to the filepath given URL and token (if applicable)
+func DownloadFile(url string, token string, filepath string) error {
```

Comment:
Optional arguments is only problem, the other problem is currently we don't have way to know which function to call (eg. `DownloadFile without token` VS `DownloadFile with token`), it's similar issue/discussion to another comment https://github.com/openshift/odo/pull/3516#discussion_r451809290

Reply:
To solve the optional arguments issue, rather than creating separate function and use shared routine, let me try variadic arguments or pass struct as arguments because they have better scalability and maintainability.